### PR TITLE
Minor project fixes and improvements

### PR DESCRIPTION
**Pull request by williamdes**

### CODE REVIEW

File: .github/workflows/docs-publish.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 name: Publish API documentation
 
+permissions:
+  contents: read
```
Comment (OndraM) [inline .github/workflows/docs-publish.yml:4]: Is this sufficient? This workflow pushes to the gh-pages branch.

Comment (williamdes) [reply .github/workflows/docs-publish.yml:4]: I think so because we push from a ssh key and not from the token obtained on the running workflow

Comment (OndraM) [reply .github/workflows/docs-publish.yml:4]: Ok, we will see :)

**Comment by OndraM** [comment]
Thanks @williamdes!

**Pull request merged**